pkg/cluster/cluster.go

```diff
@@ -211,11 +211,10 @@ func (c *Cluster) run(stopC <-chan struct{}, wg *sync.WaitGroup) {
 		if needDeleteCluster {
 			c.logger.Infof("deleting cluster")
 			c.delete()
+			go c.reportFailedStatus()
```

Comment:
> we should not delete the cluster when reconcile fails.

Reconcile only fail when it should. Create an new issue and let's discuss elsewhere.
This PR is trying to fix controller restore case. We shouldn't report failed.

> we should keep the trp so that user can see why it fails.

Operator didn't delete it.

> user should delete the tpr themselves.

Same as above